src/librustc_codegen_ssa/back/link.rs

```diff
@@ -918,7 +918,53 @@ pub fn print_native_static_libs(sess: &Session, all_native_libs: &[NativeLibrary
     }
 }
 
+// Because windows-gnu target is meant to be self-contained for pure Rust code it bundles
+// own mingw-w64 libraries. These libraries are often not compatible with mingw-w64
+// installed in the system. This breaks many cases where Rust is mixed with other languages
+// (e.g. *-sys crates).
+// We prefer system mingw-w64 libraries if they are available to avoid this issue.
+fn get_crt_libs_path(sess: &Session) -> Option<PathBuf> {
+    use std::sync::Mutex;
+
+    lazy_static::lazy_static! {
+        static ref SYSTEM_LIBS: Mutex<Option<PathBuf>> = Mutex::new(None);
+    }
+
+    let system_libs = SYSTEM_LIBS.lock().unwrap().clone();
+    if let Some(compiler_libs_path) = system_libs {
+        println!("cache: hit");
+        return Some(compiler_libs_path);
+    } else {
+        println!("cache: miss");
```

Comment:
I think that we're going to want to cache a `None` value as well if we're caching anything at all, because if spawning a process fails we don't want to fail to spawn the process each time.

Reply:
I haven't thought about it, sounds good!